include/nuttx/sched.h

```diff
@@ -412,6 +419,52 @@ struct binary_s;                    /* Forward reference
                                     /* Defined in include/nuttx/binfmt/binfmt.h */
 #endif
 
+/* This structure encapsulates all variables associated with getopt(). */
+
+struct getopt_s
```

Comment:
could we keep getopt_s and task_info_s in tls.h?

Reply:
Done